### PR TITLE
[salesforce] Update package to format_version 3.0.0

### PR DESCRIPTION
**Pull request by tommyers-elastic**
The biggest change here is the modification of the simple `rename` processors to `script` in order to get the formatting of the list the way elastic-package wants it. I'm not sold on these changes, since it adds complexity for no change in behaviour. 

@jsoriano is there any way we could make changes in the validation for cases like this to avoid the added overhead in the ingest pipelines?

### CODE REVIEW

File: packages/salesforce/data_stream/apex/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -394,11 +394,17 @@ processors:
...
     ignore_failure: true
+    lang: painless
+    source: |
+      def userType = ctx.json?.USER_TYPE;
+      if (userType != null) {
+        if (ctx.user == null) {
+          ctx.user = new HashMap();
+        }
+        ctx.user.roles = [userType];
+      }
```
[the remark this answers is not in the document]
Comment (tommyers-elastic) [reply packages/salesforce/data_stream/apex/elasticsearch/ingest_pipeline/default.yml]: we identified a simpler way using `set` and template snippets. PR updated.

[the remark this answers is not in the document]
Comment (tommyers-elastic) [reply packages/salesforce/data_stream/apex/elasticsearch/ingest_pipeline/default.yml]: but i think i prefer append...

File: packages/salesforce/data_stream/apex/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -394,11 +394,11 @@ processors:
...
     ignore_failure: true
-- rename:
-    field: json.USER_TYPE
-    target_field: user.roles
-    ignore_missing: true
+- set:
+    value: ["{{{json.USER_TYPE}}}"]
+    field: user.roles
     ignore_failure: true
+    ignore_empty_value: true
```
Comment (shmsr) [inline packages/salesforce/data_stream/apex/elasticsearch/ingest_pipeline/default.yml:401]: You also have to add a `remove` processor to remove `json.USER_TYPE`

Comment (shmsr) [reply packages/salesforce/data_stream/apex/elasticsearch/ingest_pipeline/default.yml:401]: rename = set + remove

Comment (tommyers-elastic) [reply packages/salesforce/data_stream/apex/elasticsearch/ingest_pipeline/default.yml:401]: ah good catch - thanks!

**Comment by shmsr** [comment]
LGTM.

Although, CI failed. Format required.

**Review by shmsr — APPROVED** [review]

**Pull request merged**